src/main/java/org/alfresco/heartbeat/HBDataCollectorServiceImpl.java

```diff
@@ -56,44 +83,184 @@
     public HBDataCollectorServiceImpl (boolean defaultHeartBeatState)
     {
         this.defaultHbState = defaultHeartBeatState;
+        licenseService.registerOnLicenseChange(this);
```

Comment:
Remove licenseService.registerOnLicenseChange(this) -  the DescriptorService will register this bean with license service

Reply:
Also set the this.enabled= defaultHeartBeatState